ampt comparison (#479)
nitrictech/docs: src/pages/faq/comparison/ampt.mdx

export const description = 'Comparison to Ampt'

# Ampt vs. Nitric

> Nitric is a framework that empowers developers to build complete cloud applications on their preferred cloud platform, combining infrastructure declaration and application code in one place. On the other hand, Ampt is a development platform tailored for building cloud backend applications using the JavaScript/TypeScript language, with deployment options only for AWS. Both serve the purpose of simplifying cloud application development but operate in slightly different ways and cater to different specific needs.

Reviewer: ```suggestion
Nitric is a framework that empowers developers to build complete cloud applications on their preferred cloud platform, combining infrastructure declaration and application code in one place. Ampt is a development platform tailored for rapidly building cloud applications using JavaScript or TypeScript. Both serve the purpose of simplifying cloud application development but operate in different ways and cater to different specific needs.
```

These are the main differences between the two:

> 1. **Cloud Provider Support:** Nitric is provider-agnostic, capable of deploying to multiple cloud providers such as AWS, Google Cloud, and Azure. Ampt is primarily designed for AWS services.

Reviewer: ```suggestion
1. **Cloud Provider Support:** Nitric is provider-agnostic, capable of deploying to multiple cloud providers such as AWS, Google Cloud, and Azure. Ampt is a platform, which currently runs on AWS.
```

> 2. **Language Support:** Nitric provides multiple libraries for TypeScript/JavaScript, Python, Go, C# .NET and Java, allowing developers to choose their preferred language. Currently Ampt only supports TypeScript and JavaScript.

Reviewer: ```suggestion
2. **Language Support:** Nitric provides multiple libraries for TypeScript/JavaScript, Python, Go, C# .NET and Java, allowing developers to choose their preferred language. Currently Ampt currently supports TypeScript and JavaScript.
```

Author: I think this is redundant, just need one currently

> 3. **Infrastructure Provisioning:** Nitric uses Pulumi by default for provisioning cloud resources, but also allows the use of [custom providers](/reference/providers/custom/building-custom-provider). Ampt uses a hosted provisioning engine via their online platform.

Reviewer: ```suggestion
3. **Infrastructure Provisioning:** Nitric uses Pulumi by default for provisioning cloud resources, but also allows the use of [custom providers](/reference/providers/custom/building-custom-provider). Ampt uses a hosted provisioning engine via their online platform. While resources deployed with Nitric are statically determined at deployment, Ampt is able to reallocate resources between AWS services at runtime to optimize applications.
```

> 4. **Local Simulation:** Nitric provides a fully offline local cloud simulation tool, allowing you to test your applications locally before deploying them. This is a feature that Ampt does not offer, for local development Ampt deploys and updates your application to the cloud via a sandbox environment.

Reviewer: ```suggestion
4. **Local Simulation:** Nitric provides a fully offline local cloud simulation tool, allowing you to test your applications locally before deploying them. This is a feature that Ampt does not offer. However, for local development Ampt deploys and syncs your application to the cloud via a sandbox environment, allowing for rapid testing of local changes.
```

> 5. **Open Source:** As an open-source framework, Nitric does not require a platform or subscription to deploy applications.

Reviewer 2: Would it be beneficial to add a comment about Ampt being closed source? Currently its implied, not stated explicitly.

Author: Changed

> 5. **Open Source:** As an open-source framework, Nitric does not require a platform or subscription to deploy applications, while closed-source Ampt requires sign-up and may involve additional costs.

Reviewer: ```suggestion
5. **Open Source:** As an open-source framework, Nitric does not require a platform or subscription to deploy applications. Ampt is a hosted platform which requires an account and provides both free and paid offerings.
```

import { api } from '@nitric/sdk'

const helloApi = api('main')

helloApi.get('/hello/:name', async (ctx) => {
  const { name } = ctx.req.params

  ctx.res.body = `Hello ${name}`
})
```

```javascript {{title: 'Ampt (Experimental)', label: 'hello.ts' }}
import { api } from '@ampt/api'

const helloApi = api('main').router('/hello')

helloApi.get('/:name', async (event) => {
  const { name } = event.params

  return event.status(200).body(`Hello ${name}`)
})
```

</CodeGroup>

## Differences

|                      | Nitric                                                                                       | Ampt                                     |
| -------------------- | -------------------------------------------------------------------------------------------- | ---------------------------------------- |
| Language             | Your choice                                                                                  | JavaScript / TypeScript                  |
| Lines of code        | 9                                                                                            | 9                                        |

> | Cloud Infrastructure | Inferred                                                                                     | Inferred                                 |

Reviewer: ```suggestion
| Cloud Infrastructure | Inferred, Static                                                                                     | Inferred, Dynamic                                 |
```

| Extensibility        | [Custom providers](/reference/providers/custom/building-custom-provider) can be created      | None                                     |

> | Local simulation     | Built-in local simulator with instant hot reloading and offline support                      | No, provides proxy to a deployed sandbox |

Reviewer: ```suggestion
| Local development     | Built-in local simulator with instant hot reloading and offline support                      | Active project synchronization to a cloud sandbox with a local proxy |
```

> | Cloud providers      | AWS, Azure, GCP and [Custom providers](/reference/providers/custom/building-custom-provider) | AWS                                      |

Reviewer: ```suggestion
| Cloud providers      | AWS, Azure, GCP and [Custom providers](/reference/providers/custom/building-custom-provider) | Ampt Platform (AWS)                                      |
```
